### PR TITLE
Final pipeline

### PR DESCRIPTION
**Pull request by dhruvii1205**

### CODE REVIEW

File: WBAugmenter/WBEmulator.py [hunk 1]
```diff
@@ -39,9 +39,7 @@ def __init__(self):
     self.K = 25  # K value for nearest neighbor searching
     self.sigma = 0.25  # fall off factor for KNN
     # WB & photo finishing styles
-    self.wb_photo_finishing = ['_F_AS', '_F_CS', '_S_AS', '_S_CS',
-                               '_T_AS', '_T_CS', '_C_AS', '_C_CS',
```
Comment (tejaskfx) [inline WBAugmenter/WBEmulator.py:43]: Why was this needed?

Comment (dhruvii1205) [reply WBAugmenter/WBEmulator.py:43]: Because we needed only one file as an output.

File: image_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+
+        config = InferenceConfig()
+        config.display()
+
+        # Create model object in inference mode.
+        model = modellib.MaskRCNN(mode="inference", model_dir=MODEL_DIR, config=config)
+
+        # Load weights trained on MS-COCO
+        model.load_weights("/home/dhruvi/work/ai_photo_editing/AI_image_editing/mask_rcnn_coco.h5", by_name=True)
```
Comment (tejaskfx) [inline image_segmentation.py:145]: Removed hard coded weight.

File: image_segmentation.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
+import os
```
Comment (tejaskfx) [inline image_segmentation.py:1]: This file should be removed.

File: image_segmentation_final.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
+import os
```
Comment (tejaskfx) [inline image_segmentation_final.py:1]: Rename this file , there should be one image_segmentation.py file

File: light_calculation.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        top_left_light_percentage = lightPercentage.div_light_percentage(topleft_filepath)
+        # print(top_left_light_percentage.shape)
+        return top_left_light_percentage
+
+    def main(self, out_path):
+
+        # make directory paths
+        div_image_path = out_path + '/divided_image'
+        brightness_only_path = out_path + '/brightness_only_image'
+
```
Comment (tejaskfx) [inline light_calculation.py:39]: Remove comments from code.

File: main.py [hunk 6]
```diff
@@ -3,46 +3,111 @@
 import argparse
 import numpy as np
 from PIL import Image
-from exposure import ExposureCorrection
-from colour_correction import ColorCorrection
-from contrast_correction import Adjustment
+# from exposure import ExposureCorrection
```
Comment (tejaskfx) [inline main.py:6]: remove comments from code.

File: main.py [hunk 7]
```diff
@@ -3,46 +3,111 @@
...
+# from contrast_correction import Adjustment
 from WBAugmenter import WBEmulator as wbAug
+from wbAug import WhiteBalance
+from image_enhance import Enhancer
+from demo_single_image import DeepWhiteBalance
+from light_calculation import lightPercentage
+from image_segmentation_final import ImageSegmentation
 
-def pipeline(img):
+def pipeline(img_path, out_path):
```
Comment (tejaskfx) [inline main.py:17]: remove comments.

File: main.py [hunk 8]
```diff
@@ -3,46 +3,111 @@
...
+    light_percentage = lightPercentage(img_path, out_path)
+    value = light_percentage.main(out_path)
+    print('value: ',value)
+
+    white_balance = WhiteBalance(img_path, out_path)
+    enhanced_image_path = white_balance.main()    
+    print('enhanced_image_path_wbaug : ',enhanced_image_path)
+
+    if value < 30:
+
```
Comment (tejaskfx) [inline main.py:37]: remove comments

File: main.py [hunk 9]
```diff
@@ -3,46 +3,111 @@
...
     args = vars(parser.parse_args())
-    img = cv2.imread(args['image'])
+    img_dir = args['image_dir_path']
+    out = args['output_dir_path']
+
+    for img in os.listdir(img_dir):
+
+        img = os.path.join(img_dir, img)
+        pipeline(img, out)
+    # pipeline(img_dir, out)
```
Comment (tejaskfx) [inline main.py:97]: Remove this if not needed.

File: main.py [hunk 10]
```diff
@@ -3,46 +3,111 @@
...
+
+        suffix = ('T_AS.jpg', 'T_AS_deep_wb_enhanced.jpg', 'T_AS_deep_wb.jpg', 'brightness3.jpg', 'image_1.jpg', 'image_2.jpg')
+        # suffix1 = ('T_AS.jpg', 'T_AS_deep_wb_enhanced.jpg', 'brightness3.jpg', 'image_1.jpg', 'image_2.jpg')
```
Comment (tejaskfx) [inline main.py:100]: Remove comment.

File: main.py [hunk 11]
```diff
@@ -3,46 +3,111 @@
...
+        for filename in os.listdir(out):
+            if filename.endswith(suffix):
+                os.remove(os.path.join(out, filename))
+            else:
+                continue
+
 
-    # pipeline(img)
-    enhanced_img = pipeline(img)
+    # _ ,file_name = os.path.split(img_dir)
```
Comment (tejaskfx) [inline main.py:108]: remove comment.

File: main.py [hunk 12]
```diff
@@ -3,46 +3,111 @@
...
```
Comment (tejaskfx) [inline main.py:109]: Remove comments.

File: params/encoderBias.npy
Comment (tejaskfx) [inline params/encoderBias.npy:1]: Remove .npy files, add entry in README if needed so that we can add this later when testing.

**Review by tejaskfx — CHANGES_REQUESTED** [review]

File: image_segmentation.py (hunk 2 above)
Comment (dhruvii1205) [reply image_segmentation.py:145]: removed.

File: image_segmentation.py (hunk 3 above)
Comment (dhruvii1205) [reply image_segmentation.py:1]: removed.

File: image_segmentation_final.py (hunk 4 above)
Comment (dhruvii1205) [reply image_segmentation_final.py:1]: done.

File: light_calculation.py (hunk 5 above)
Comment (dhruvii1205) [reply light_calculation.py:39]: done.

File: main.py (hunk 6 above)
Comment (dhruvii1205) [reply main.py:6]: done.

File: main.py (hunk 7 above)
Comment (dhruvii1205) [reply main.py:17]: done.

File: main.py (hunk 8 above)
Comment (dhruvii1205) [reply main.py:37]: done.

File: main.py (hunk 9 above)
Comment (dhruvii1205) [reply main.py:97]: done.

File: main.py (hunk 10 above)
Comment (dhruvii1205) [reply main.py:100]: done

File: main.py (hunk 11 above)
Comment (dhruvii1205) [reply main.py:108]: done.
